shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/common/statement/dml/MergeStatement.java

```diff
@@ -37,4 +41,17 @@
     private TableSegment source;
     
     private ExpressionSegment expr;
+    
```

Comment:
@tristaZero Hi, do you think it is necessary to add the 'setAssignment' and 'where' in MergeStatement?
Related rule as below:
```
mergeSetAssignmentsClause
    : SET mergeAssignment (COMMA_ mergeAssignment)*
    ;

whereClause
    : WHERE expr
    ;
```

Reply:
Hi,

First, `mergeSetAssignmentsClause` is OK for me, since it is like a definition with a group of `mergeSetAssignment`. Therefore a `merge` main definition will contain some of the keywords and some of the child group definitions, and each group definition is made up of its child definitions. 
On the other hand, you can still put all specific definitions in the main clause, which does not have any group definitions.

Which one is our best choice? I suppose we need to add another factor, i.e., `SQLStatement`. Suppose we want to visit out some specific `statement`, it is necessary to estimate which type of SQL definition of `g4` file is more reasonable and straightforward for visitors.

That's my opinion and hope it can give you some hints for the following issues.

Second, actually, a more fitting member putting into `MergeStatement`, I guess, is the `UpdateStatement` containing `setAssignment` and `where`. The reason is that `OracleUpdateStatement` has existed and owned the entire and complete definition about `update set where`. 
But that will make it more challenging to do unit tests. I suggest you can have a try, and if that frustrates you too much, you can put it aside. Yep, it is optional. :)

@ThanoshanMV 
Hope you can succeed in GSoC 2021 and learn more from this community.  Also @wgy8283335 and I will provide our mentoring as much as we can.